pilot/pkg/proxy/envoy/v2/eds_test.go

```diff
@@ -828,6 +826,7 @@ func addOverlappingEndpoints(server *bootstrap.Server) {
 		},
 	})
 	server.EnvoyXdsServer.Push(&model.PushRequest{Full: true})
+	time.Sleep(1 * time.Second)
```

Comment:
Donot we have wait time in `TestEDSServiceResolutionUpdate`?

Reply:
If you are referring to `adscConn := adsConnectAndWait(t, 0x0a0a0a0a)` - It comes out on first push that happens in `initLocalPilotTestEnv` I think. The second `addEdsCluster` will have another push that is not waited up on. That is what is actually making these tests that add additional cluster after `initLocalPilotTestEnv` is done is failing.

Reply:
And another point, those waits just wait for a "cds", "lds" , "eds" - but not specific eds response as an example. Since we have many clusters that we set up in `initLocalPilotTestEnv` those waits are not really helping especially if we add additional clusters. That is why I was thinking that we should just add clusters that we care about in the test setup not the entire world.